### ISSUE TITLE
depthPP values will cause salinityPP to fail over

### ISSUE THREAD

**Issue by ggalibert** [aodn/imos-toolbox#375]
@sspagnol says:

Have a case where from depthPP some values are created that will cause salinityPP to fail over. Can we add something like

computedDepth(computedDepth < 5) = NaN;

to depthPP.m at around line 560 (before the data is added to sample structure). Value 5 chosen as that is the cutoff in gsw_p_from_z that causes the issue.

**Comment by ggalibert** [comment]
@sspagnol continues:

And to follow on - in gsw_p_from_z the z is depth negative so think the line in depthPP should be

computedDepth(computedDepth < -5) = NaN;

or push this operation into salinityPP after line 113 so it would look like

depth = sam.(depthType){depthIdx}.data;
depth(depth < -5) = NaN;

**Comment by ggalibert** [comment]
@sspagnol can you please send me a copy of datasets/ddb involved so that I can reproduce your problem?

**Comment by ggalibert** [comment]
```gsw_p_from_z()``` **is not used** in depthPP.m so not too sure about your comment above.

computedDepth is used to document the IMOS parameter DEPTH, which is positive downward. Indeed, computedDepth can be obtained from ```- gsw_z_fromp()``` calls where the resulting z is positive downwards (hence the sign to change convention).

**Comment by sspagnol** [comment]
The issue I had was with a data set where depthPP generated some depth values > 5 (either due to a dodgy pressure reading at the end of a deployment or whatever). When preprocessing step was followed by a salinityPP, the toolbox would fall over because salinityPP would call gsw_p_from_z which has

```matlab
if any(z > 5)
   error('gsw_p_from_z: The input z is generally negative and values must be less than 5 m.')
end
```
I just think in salinityPP, depth should be in valid range only (i.e. depth(depth < -5) = NaN;) and also set qcflag as bad for those values. 

At least this why when debugging I can examine the sample_data structure(s) to look at why depthPP has done what it has done.

**Comment by ggalibert** [comment]
Ok, thanks for the data sets, now makes sense.

Will add line 128 of salinityPP.m:
```Matlab
      % any depth values <= -5 are discarded (reminder, depth is
      % positive down), this allow use of gsw_p_from_z without error.
      depth(depth <= -5) = NaN;
```

Check your in/out water times for ITFTIS-1602 - Seabird SBE37SM-RS232 SN=03709097 @31m (SBE3790971607.cnv) in your ddb, the values are wrong and flag most of the timeserie.

**Comment by ggalibert** [comment]
Fixed in commit https://github.com/aodn/imos-toolbox/commit/8dbfc14611be4618c7e8242e424a0c372394532d, included in v2.5.19.
